pkg/controller/elasticsearch/user/predefined.go

```diff
@@ -83,18 +85,26 @@ func reconcileInternalUsers(
 	existingFileRealm filerealm.Realm,
 	passwordHasher cryptutil.PasswordHasher,
 ) (users, error) {
+	users := users{
+		{Name: ControllerUserName, Roles: []string{SuperUserBuiltinRole}},
+		{Name: PreStopUserName, Roles: []string{ClusterManageRole}},
+		{Name: ProbeUserName, Roles: []string{ProbeUserRole}},
+		{Name: MonitoringUserName, Roles: []string{RemoteMonitoringCollectorBuiltinRole}},
+		{Name: DiagnosticsUserName, Roles: []string{DiagnosticsUserRoleV85}},
+	}
+	ver, err := version.Parse(es.Spec.Version)
+	if err != nil {
+		return nil, fmt.Errorf("while parsing Elasticsearch version (%s): %w", es.Spec.Version, err)
+	}
+	if ver.LT(version.From(8, 5, 0)) {
+		users[4].Roles = []string{DiagnosticsUserRoleV7}
```

Comment:
Using an index (`4`) while we want to use a role for a specific user feels error prone to me. If the order of users is changed in a later PR, it would assign the role to another user without being noticed in unit tests in all the cases.

Reply:
I added a func to replace the roles. lmk if this is the direction you intended.